pkg/scheduler/event_handler.go

```diff
@@ -202,6 +206,17 @@ func (s *Scheduler) enqueueAffectedBindings(oldCluster, newCluster *clusterv1alp
 
 		var affinity *policyv1alpha1.ClusterAffinity
 		if placementPtr.ClusterAffinities != nil {
+			// If the status of binding obtained from the informer is not the latest,
+			// try to get binding with client to prevent the loss of cluster update event
+			if binding.Status.SchedulerObservedGeneration != binding.Generation {
```

Comment:
In case of  `binding.Status.SchedulerObservedGeneration != binding.Generation`, there is also a possibility that the RB hasn't been scheduled. I doubt that cluster update event need to rely on the scheduled state.

Reply:
> I doubt that cluster update event need to rely on the scheduled state.

This is mainly because our `SchedulerObservedAffinityName` field is in status, to ensure it is up to date.

Reply:
Yes, we need to know what conditions to retry or access karmada-apiserver to get the latest binding object. The reason why the retry is not performed is that the binding object that has been added to the re-scheduling queue is added again.

Reply:
If `binding.Status.SchedulerObservedGeneration != binding.Generation` is true, it might mean:
1. The ResourceBinding hasn't been scheduled.
2. The ResourceBinding has been scheduled, but the status hasn't been cached.

Refresh the ResourceBinding can only resolve `2`, how to handle case `1`? Note that, the `binding.Status.SchedulerObservedAffinityName` would be empty.

Reply:
Assuming we have 50 member clusters, if there are currently 10 ResourceBindings that have not been scheduled, how many ResourceBindings will be added to the queue once all clusters trigger an update event?

 I guess it would be 500, right?

Reply:
The same key added to the queue will be eliminated. There should be less than 500 keys.

Reply:
ohh